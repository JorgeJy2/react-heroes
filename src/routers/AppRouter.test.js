import { mount } from "enzyme";
import { AuthContext } from "../components/auth/AuthContext";
import { AppRouter } from "./AppRouter";

describe('test in app route', () => {

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false
    }
  }

  test('should show login if user not auth', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('LoginScreen');
  });

  test('should show marvel component if user auth', () => {
    contextValue.user = {
      logged: true,
      name: 'jorge'
    }
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>);
    expect(wrapper.find('.navbar').exists()).toBeTruthy();
  });


});