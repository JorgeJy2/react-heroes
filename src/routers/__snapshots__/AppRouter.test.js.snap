// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in app route should show login if user not auth 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
       Login
    </button>
  </div>
</div>
`;
